### PR TITLE
Add lowering support for math::AbsIOp

### PR DESCRIPTION
**Pull request by aviator19941**
There is no lowering support for math::AbsIOp, so if the operand is an integer type, it will fail to lower to math::AbsFOp since the op operand #0 must be floating-point-like.

### CODE REVIEW

File: lib/Conversion/TorchToLinalg/Uncategorized.cpp [hunk 1]
```diff
@@ -424,8 +424,11 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
         b.create<arith::ConstantOp>(loc, b.getFloatAttr(floatDtype, 0));
     return createEqual(b, loc, floatDtype, self, zero);
   }
-  if (isa<AtenAbsOp>(op))
+  if (isa<AtenAbsOp>(op)) {
+    if (payloadArgs[0].getType().isa<IntegerType>())
+      return b.create<math::AbsIOp>(loc, payloadArgs[0]);
     return b.create<math::AbsFOp>(loc, payloadArgs[0]);
+  }
```
Comment (vivekkhandelwal1) [inline lib/Conversion/TorchToLinalg/Uncategorized.cpp:431]: Are you sure that apart from integer or float it doesn't take any other element type?

Comment (aviator19941) [reply lib/Conversion/TorchToLinalg/Uncategorized.cpp:431]: I am not sure, but I will check. All I know is that before it was only lowering to AbsFOp.

Comment (aviator19941) [reply lib/Conversion/TorchToLinalg/Uncategorized.cpp:431]: MLIR only has math::AbsIOp/AbsFOp here https://mlir.llvm.org/docs/Dialects/MathOps/, so it should only take integer or float element types.

**Review by newling — COMMENTED** [review]
LGTM, just the small test comment

File: projects/pt1/python/torch_mlir_e2e_test/test_suite/elementwise.py [hunk 2]
```diff
@@ -2127,14 +2127,36 @@ def forward(self, a):
...
+        None,
+        ([-1, -1, -1], torch.int64, True),
+    ])
+    def forward(self, a):
+        return torch.abs(a)
+
+
+@register_test_case(module_factory=lambda: ElementwiseAbsIntModule())
+def ElementwiseAbsIntModule_basic(module, tu: TestUtils):
+    module.forward(tu.randint(3, 4, 5, low=-1, high=1))
```
[the remark this answers is not in the document]
Comment (aviator19941) [reply projects/pt1/python/torch_mlir_e2e_test/test_suite/elementwise.py]: Yeah that makes sense, I changed the tests to specify one negative value always.

**Pull request merged**